### PR TITLE
Feature: Ftm time to harvest

### LINKED ISSUE #155: Harvest FTM based on time since last harvest
Add time since last harvest support for FTM. Right now we're harvesting on a daily schedule that will not rerun for a vault if its harvest fails. We should move to a similar model to ETH where we use ftmscan api to check the time since the last harvest for each vault/strategy and harvest if it exceeds a time threshold.

Completion of this issue should include unit tests for any new functions / branches in existing functions.

Example code for implementation of ETH is in `scripts/eth_harvest.py`

### PR DESCRIPTION
**Pull request by SHAKOTN**
**Done:**
1. Add support for fetching latest harvest times from ftmscan for FTM chain
2. Add support for FTM in `is_time_to_harvest()` method
3. Move `safe_harvest` func to separate module
4. Cover everything with tests

Closes #155

### CODE REVIEW

File: src/general_harvester.py [hunk 1]
```diff
@@ -51,7 +52,7 @@ def __init__(
         self.web3 = web3
         self.keeper_key = keeper_key
         self.keeper_address = keeper_address
-        self.keeper_acl = self.web3.eth.contract(
+        self.keeper_acl: Contract = self.web3.eth.contract(
```
Comment (btcookies) [inline src/general_harvester.py:55]: why are we typing this member?

File: src/web3_utils.py [hunk 2]
```diff
@@ -127,26 +131,32 @@ def get_last_harvest_times(
...
     Returns:
         dict: Dictionary of strategy addresses and their latest harvest timestamps.
     """
-    if etherscan_key is None:
-        etherscan_key = get_secret("keepers/etherscan", "ETHERSCAN_TOKEN")
+    if chain == Network.Ethereum:
+        api_key = get_secret("keepers/etherscan", "ETHERSCAN_TOKEN")
+        url = "https://api.etherscan.io/api"
+    elif chain == Network.Fantom:
+        api_key = get_secret("keepers/etherscan", "FTMSCAN_TOKEN")
```
Comment (btcookies) [inline src/web3_utils.py:143]: change the first param here to keepers/ftmscan, i'll create secret

File: src/general_harvester.py (hunk 1 above)
Comment (SHAKOTN) [reply src/general_harvester.py:55]: For better type inference :)

Comment (btcookies) [reply src/general_harvester.py:55]: But not typing the base_usd_oracle? Seemed inconsistent

Comment (SHAKOTN) [reply src/general_harvester.py:55]: Ok, I can do that. Was just working with keeper_acl, so decided to type it properly

Comment (SHAKOTN) [reply src/general_harvester.py:55]: Done

File: src/web3_utils.py (hunk 2 above)
Comment (SHAKOTN) [reply src/web3_utils.py]: Done

**Review by btcookies — APPROVED** [review]
LGTM

**Pull request merged**